engines/io_uring.c

```diff
@@ -640,6 +683,22 @@ static int fio_ioring_init(struct thread_data *td)
 	ld->iovecs = calloc(td->o.iodepth, sizeof(struct iovec));
 
 	td->io_ops_data = ld;
+
+	/*
+	 * Initialize hiprio_percent mechanisms if relevant
+	 */
+	if (!(fio_option_is_set(to, ioprio) ||
```

Comment:
With the above suggestion, this can go away and you're left with just checking for the conflicting options. Ditto in libaio.